invenio_oauthclient/handlers/base.py

```diff
@@ -46,14 +47,63 @@
 )
 
 
+def _role_needs_update(role_obj, new_role_dict):
+    """Checks if role needs to be updated."""
+    if role_obj.name != new_role_dict.get("name") or role_obj.description != new_role_dict.get("description"):
+        return True
+    return False
+
+
+def _create_or_update_groups(account_groups):
+    """Creates the roles based on the groups provided."""
+    roles_id = []
+    for group in account_groups:
+        existing_role = current_datastore.find_role_by_id(group["id"])
+        if existing_role and existing_role.is_managed:
+            current_app.logger.exception(
+                f'Error while syncing roles: A managed role with id: ${group["id"]} already exists'
+            )
+            continue
+        existing_role_by_name = current_datastore.find_role(group["name"])
```

Comment:
can we check if an `index` can be added here when the task to import groups in CDS is done?